tests/test.rs

```diff
@@ -57,7 +60,7 @@ fn run(path: &Path) {
     assert!(status.success());
 }
 
-fn read_expected(path: &Path, compiler: &str, compiler_ver: &str, format: &str) -> String {
+fn read_expected(path: &Path, append: &str, compiler: &str, compiler_ver: &str, format: &str) -> String {
```

Comment:
This function is slowly becoming very ugly, luckily it's a test so we don't care too much.

We could split it in two though, one function that only does:
```
     let mut f = File::open(path.join(&name)).expect(format!("{} file not found", name).as_str());
     let mut s = String::new();
     f.read_to_string(&mut s).unwrap();
     s
```

and the other which builds the path and then calls the first.

In test_integration we can call the second, in your test you can call the first with a hardcoded path.